Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -858,8 +1039,67 @@ fileprivate extension Compiler.ByteCodeGen {
       }
       return
     }
-    let consumer = try ccc.generateConsumer(options)
-    builder.buildConsume(by: consumer)
+
+    let updatedCCC: DSLTree.CustomCharacterClass
+    if optimizationsEnabled {
+      updatedCCC = ccc.coalescingASCIIMembers(options)
+    } else {
+      updatedCCC = ccc
+    }
+    let filteredMembers = updatedCCC.members.filter({!$0.isOnlyTrivia})
```

Comment:
Quick note: I think it makes sense for this to be a single operation that will recursively traverse and produce a code-gen-relevant representation. That will also help with testing as we can test that the flattened output is properly associated and has the members we expect. It will also help divide out linearizing the set structure into the instruction stream from removing consumer functions themselves.

E.g.
```
extension DSLTree.CustomCharacterClass {
  enum CodeGenClassMember {
    case bitset: ASCIIBitset
    case consumerFunction: () -> () ...
    case range: ...
    case character: Character
    case scalar: Unicode.Scalar
  }

  enum Operation {
    case union, intersection, ...
    case endOfOps
  }

  func codeGenRepresentation() -> [(CodeGenClassMember, Operation)]
}
```